h/static/scripts/group-forms/components/test/CreateGroupForm-test.js

```diff
@@ -3,20 +3,83 @@ import { mount } from 'enzyme';
 import CreateGroupForm from '../CreateGroupForm';
 
 describe('CreateGroupForm', () => {
-  const createWrapper = () => mount(<CreateGroupForm />);
+  /* Return true if counterEl is in its error state. */
+  const counterIsInErrorState = counterEl => {
+    return (
+      counterEl.hasClass('text-red-error') && counterEl.hasClass('font-bold')
+    );
+  };
+
+  /* Return true if component is in its error state. */
+  const componentIsInErrorState = component => {
+    return component.prop('error') !== undefined;
+  };
+
+  const getElements = wrapper => {
+    return {
+      name: {
+        counterEl: wrapper.find('[data-testid="charcounter-name"]'),
+        fieldComponent: wrapper.find('Input[data-testid="name"]'),
+        fieldEl: wrapper.find('input[data-testid="name"]'),
+      },
+      description: {
+        counterEl: wrapper.find('[data-testid="charcounter-description"]'),
+        fieldComponent: wrapper.find('Textarea[data-testid="description"]'),
+        fieldEl: wrapper.find('textarea[data-testid="description"]'),
+      },
+    };
+  };
+
+  const createWrapper = () => {
+    const wrapper = mount(<CreateGroupForm />);
+    const elements = getElements(wrapper);
+    return { wrapper, elements };
+  };
```

Comment:
I don't think I've seen this in any of the existing component tests that I looked at, but I found that my tests kept repeating this kind of code:

```javascript
const wrapper = createWrapper();
const counterEl = wrapper.find('[data-testid="charcounter-name"]');
const fieldEl = wrapper.find('input[data-testid="name"]');
...
```

...so I changed `createWrapper()` to also find and return all the sub-components and sub-elements so that tests can just do this:

```javascript
const { counterEl, fieldEl } = createWrapper();
```

Reply:
This is fine. Writing ad-hoc helpers to navigate a `wrapper` is common:

```js
function getByTestId(wrapper, testId) {
  return wrapper.find(`[data-testid="${testId}"]`);
}

const counter = getByTestId(wrapper, "counter");
```